src/main/java/io/harness/cf/client/api/AuthService.java

```
@@ -7,62 +7,110 @@
 import io.harness.cf.api.ClientApi;
 import io.harness.cf.model.AuthenticationRequest;
 import io.harness.cf.model.AuthenticationResponse;
+
 import java.util.concurrent.TimeUnit;
+
 import lombok.NonNull;
 import lombok.extern.slf4j.Slf4j;
 
 @Slf4j
 public class AuthService extends AbstractScheduledService {
 
-  protected final String apiKey;
-  protected final CfClient cfClient;
-  protected final ClientApi defaultApi;
-  protected final int pollIntervalInSec;
+    protected final String apiKey;
+    protected final CfClient cfClient;
+    protected final ClientApi defaultApi;
+    protected final int pollIntervalInSec;
+    protected final AuthCallback callback;
+
+    public AuthService(
+
+            final ClientApi defaultApi,
+            final String apiKey,
+            final CfClient cfClient,
+            final int pollIntervalInSec,
+            final AuthCallback callback
+    ) {
+
+        this.apiKey = apiKey;
+        this.cfClient = cfClient;
+        this.callback = callback;
+        this.defaultApi = defaultApi;
+        this.pollIntervalInSec = pollIntervalInSec;
+    }
 
-  public AuthService(
-      ClientApi defaultApi, String apiKey, CfClient cfClient, int pollIntervalInSec) {
+    @Override
+    protected void runOneIteration() throws Exception {
 
-    this.defaultApi = defaultApi;
-    this.apiKey = apiKey;
-    this.cfClient = cfClient;
-    this.pollIntervalInSec = pollIntervalInSec;
-  }
+        if (isNullOrEmpty(apiKey)) {
 
-  @Override
-  protected void runOneIteration() throws Exception {
+            final Exception error = new CfClientException("SDK key cannot be empty");
+            failure(error);
+            throw error;
```

Should we throw the error or just log it?